app/src/main/java/com/github/ashutoshgngwr/noice/fragment/RandomPresetFragment.kt

```diff
@@ -0,0 +1,54 @@
+package com.github.ashutoshgngwr.noice.fragment
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import com.github.ashutoshgngwr.noice.MediaPlayerService
+import com.github.ashutoshgngwr.noice.R
+import com.github.ashutoshgngwr.noice.databinding.RandomPresetFragmentBinding
+import com.github.ashutoshgngwr.noice.sound.Sound
+
+class RandomPresetFragment : Fragment() {
+
+  private lateinit var binding: RandomPresetFragmentBinding
+
+  override fun onCreateView(
+    inflater: LayoutInflater, container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View {
+    binding = RandomPresetFragmentBinding.inflate(inflater, container, false)
+    return binding.root
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+
+    binding.playPresetButton.setOnClickListener {
+      val tag = when (binding.presetType.checkedRadioButtonId) {
+        R.id.preset_type__focus -> Sound.Tag.FOCUS
+        R.id.preset_type__relax -> Sound.Tag.RELAX
+        else -> null
+      }
+
+      val intensity = when (binding.presetIntensity.checkedRadioButtonId) {
+        R.id.preset_intensity__light -> SoundLibraryFragment.RANGE_INTENSITY_LIGHT
+        R.id.preset_intensity__dense -> SoundLibraryFragment.RANGE_INTENSITY_DENSE
+        else -> SoundLibraryFragment.RANGE_INTENSITY_ANY
+      }
+      MediaPlayerService.playRandomPreset(requireContext(), tag, intensity)
+      //We'll add later a if player is playing then show else there was an error
+      Toast.makeText(requireContext(), "Playing Random!", Toast.LENGTH_SHORT).show()
```

Comment:
I think clicking the play button should just play a random preset regardless if the player was already playing. The logic remains the same. MediaPlayerService internally stops the existing playback before starting a random (or any) preset. There's no error scenario here per-say.

Reply:
True! But I explained myself bad there, let me try again:
Check if is playing like the previous action did its work. If MediaPlayer is not playing (Thing that should never happen after executing `MediaPlayerService.playRandomPreset()`) then show an error. 

Most likely an error will never happen, but just to be sure!